solutions/usaco-1064.mdx

```diff
@@ -108,4 +108,111 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class StuckInARut {
+	static int[] xCoordinates;
+	static int[] yCoordinates;
+
+	// sorts eastward cows by their y value
+	static class EastComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			if (yCoordinates[first] < yCoordinates[second]) {
+				return -1;
+			}
+			return 1;
+		}
+	}
+
+	// sorts northward cows by their x value
+	static class NorthComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			if (xCoordinates[first] < xCoordinates[second]) {
+				return -1;
+			}
+			return 1;
+		}
+	}
+
+	public static void main (String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		int n = Integer.parseInt(br.readLine());
+
+		List<Integer> eastCows = new ArrayList<>();
+		List<Integer> northCows = new ArrayList<>();
+		xCoordinates = new int[n];
+		yCoordinates = new int[n];
+
+		for (int x = 0; x < n; x++) {  // read input
```

Comment:
maybe change this to i? x could be a bit misleading since this problem involves x coordinates